src/MongoDB.Driver.Legacy/MongoServerSettings.cs

```diff
@@ -727,10 +745,13 @@ public static MongoServerSettings FromClientSettings(MongoClientSettings clientS
             serverSettings.ApplicationName = clientSettings.ApplicationName;
             serverSettings.ClusterConfigurator = clientSettings.ClusterConfigurator;
             serverSettings.Compressors = clientSettings.Compressors;
+#pragma warning disable 618
             serverSettings.ConnectionMode = clientSettings.ConnectionMode;
+#pragma warning restore 618
             serverSettings.ConnectTimeout = clientSettings.ConnectTimeout;
 #pragma warning disable 618
             serverSettings.Credentials = clientSettings.Credentials;
+            serverSettings.DirectConnection = clientSettings.DirectConnection; //todo: not safe
```

Comment:
this assignment is not safe, because the `clientSettings.DirectConnection` is not nullable now. But the state which we want to assign(to serverSettings._directConnection) can be nullable. Let's discuss it